Milestone #4: Processing Date in Models

Anjacodes opened the pull request:
In this milestone, I:

* Created the following custom methods:
   * Users
        -> A method that returns the 3 most recent posts for a given user.
   * Posts
        -> A method that updates the posts counter for a user.
        -> A method that returns the 5 most recent comments for a given post.
   * Comments
        -> A method that updates the comments counter for a post.
   * Likes
        -> A method that updates the likes counter for a post.

* Created a method called current_user in your ApplicationController which returns the first user from a database.

Anjacodes committed:
Add Methods to Models

Anjacodes committed:
Fix Linters

Anjacodes committed:
Edit Model Methods

Anjacodes committed:
Linters!

MahmoudBakr23 commented on app/models/user.rb:
```diff
@@ -0,0 +1,8 @@
+class User < ApplicationRecord
+  has_many :posts
+  has_many :comments
+
```
- [ ] Please add the correct association between the `user` and `like` models
`has_many :likes, foreign_key: 'author_id'`

MahmoudBakr23 commented on app/models/comment.rb:
```diff
@@ -0,0 +1,9 @@
+class Comment < ApplicationRecord
+  belongs_to :user
+  belongs_to :post, foreign_key: :post_id
```
**OPTIONAL**
- Please remove the `foreign_key: :post_id` from this association because it's not necessary since the `Comment` model already knows that its data belongs to a `Post` model, so it won't make any difference if we specify the foreign_key 😉

MahmoudBakr23 commented on app/models/comment.rb:
```diff
@@ -0,0 +1,9 @@
+class Comment < ApplicationRecord
+  belongs_to :user
```
- [ ] Please add the correct foreign keys to your database tables in order to point at a `user` as an `author`
`belongs_to :author, class_name: 'User'`

MahmoudBakr23 commented on app/models/like.rb:
```diff
@@ -0,0 +1,9 @@
+class Like < ApplicationRecord
+  belongs_to :post
+  belongs_to :user
```
- [ ] Please add the correct foreign keys to your database tables in order to point at a `user` as an `author`
`belongs_to :author, class_name: 'User'`

MahmoudBakr23 commented on app/models/post.rb:
```diff
@@ -0,0 +1,14 @@
+class Post < ApplicationRecord
+  belongs_to :user
```
- [ ] Please add the correct foreign keys to your database tables in order to point at a `user` as an `author`
`belongs_to :author, class_name: 'User'`

MahmoudBakr23 commented on app/models/user.rb:
```diff
@@ -0,0 +1,8 @@
+class User < ApplicationRecord
+  has_many :posts
```
- [ ] Please add the correct foreign keys to your database tables in order to point at a `user` as an `author`
`has_many :posts, foreign_key: 'author_id'`

MahmoudBakr23 commented on app/models/post.rb:
```diff
@@ -0,0 +1,14 @@
+  belongs_to :user
+  has_many :comments
+  has_many :likes
+
```
**OPTIONAL**
- Please consider using the `after_save` method to call the `update_counter` every time the model changes!

`after_save :update_counter`

MahmoudBakr23 reviewed (changes requested):
Hello @Anjacodes 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Great job implementing the logic for increasing the counters

However, there is always room for improvement.

> Keep in mind that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄


 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
_Please check the comments under the review._

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Anjacodes committed:
Implement Suggestions From CR

Anjacodes commented:
Hi @MahmoudBakr23,
Thank you for all the feedback! I guess I did a bit of a sloppy job there 🙈 .
I have implemented all the suggested changes but when I went to resubmit and request another review, I only had the option to submit, so I did. Would you anyway be able to review and see if I implemented your suggestions correctly?!
Thank you for all your help!
